src/Symfony/Contract/Service/ResettableInterface.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Contract\Service;
+
+/**
+ * Provides a way to reset an object to its initial state.
+ *
+ * When calling the "reset()" method on an object, it should be put back to its
+ * initial state. This usually means clearing any internal buffers and forwarding
+ * the call to internal dependencies. All properties of the object should be put
+ * back to the same state it had when it was first ready to use.
+ *
+ * This method could be called, for example, to recycle objects that are used as
+ * services, so that they can be used to handle several requests in the same
+ * process loop (note that we advise making your services stateless instead of
+ * implementing this interface when possible.)
+ */
+interface ResettableInterface
```

Comment:
In the Symfony world, we would say `ResetInterface` (I don't think we have any -able interface right now in Symfony). In the ZF world, it would be `Resettable`. I propose to rename to `ResetInterface` here for consistency with our practices. Any better ideas?

Reply:
@fabpot `Symfony\Component\HttpKernel\RebootableInterface`, `Symfony\Component\HttpKernel\TerminableInterface`, `Symfony\Component\Cache\PruneableInterface`, `Symfony\Component\Cache\ResettableInterface`...